lib/trino-filesystem-manager/src/main/java/io/trino/filesystem/manager/FileSystemConfig.java

```diff
@@ -71,15 +75,23 @@ public FileSystemConfig setNativeGcsEnabled(boolean nativeGcsEnabled)
         return this;
     }
 
-    public boolean isCacheEnabled()
+    public FileSystemCacheMode getCacheMode()
     {
-        return cacheEnabled;
+        return cacheMode;
     }
 
-    @Config("fs.cache.enabled")
-    public FileSystemConfig setCacheEnabled(boolean enabled)
+    @Config("fs.cache.mode")
+    public FileSystemConfig setCacheMode(FileSystemCacheMode enabled)
     {
-        this.cacheEnabled = enabled;
+        this.cacheMode = enabled;
         return this;
     }
+
+    public enum FileSystemCacheMode
+    {
+        ENABLED, /** all nodes **/
```

Comment:
ENABLED -> ALL_NODES

COORDINATOR_ONLY -> COORDINATOR_NODE

Why? COORDINATOR_ONLY is also a form of enablement.